src/auth/auth.go

```diff
@@ -0,0 +1,99 @@
+package auth
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/jinzhu/gorm"
+
+	"github.com/json9512/mediumclone-backendwithgo/src/users"
+)
+
+type loginCredentials struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+type errorResponse struct {
+	Msg string `json:"message"`
+}
+
+// AddRoutes adds HTTP Methods for the /posts endpoint
+func AddRoutes(router *gin.Engine, db *gorm.DB) {
+	router.POST("/login", func(c *gin.Context) {
+		var userCredentials loginCredentials
+		if err := c.BindJSON(&userCredentials); err != nil {
+			c.JSON(
+				http.StatusBadRequest,
+				&errorResponse{
+					Msg: "Authentication failed. Invalid data type.",
+				},
+			)
+			return
+		}
+
+		// Save access token and resfresh token [Testing for now]
+		var user users.User
+		dbResult := db.Where("email = ?", userCredentials.Email).Find(&user)
```

Comment:
요런 where email 같은 친구들은 자주 사용될 가능성 있으니 다른 함수로 빼도 좋을듯.

그리고 유저를 찾는 것이기에 return은 유져여야할 것 같음. 아 근데 보니 객체네 `dbResult.Error` 있는걸 보니. gorm 잘 몰라성....알아서 적절히 하세용
```suggestion
		user := db.Where("email = ?", userCredentials.Email).Find(&user)
```